--- setup.py        2012-02-03 13:15:50.000000000 -0800
+++ setup.py        2012-02-03 13:15:50.000000000 -0800
@@ -44,9 +44,8 @@
         arrayobject_h_include = [os.path.join(sys.prefix,
                                  "Lib/site-packages/numpy/core/include")]
     else:
-        arrayobject_h_include = [os.path.join(sys.prefix,
-                                "lib/python%s.%s/site-packages/numpy/core/include"
-                                % sys.version_info [:2])]
+        import numpy.distutils.misc_util
+        arrayobject_h_include = numpy.distutils.misc_util.get_numpy_include_dirs()
 
 math_libraries = []
 if sys.platform != 'win32':

